hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/statemachine/background/BlockDeletingService.java

```diff
@@ -374,16 +394,17 @@ public ContainerBackgroundTaskResult deleteViaSchema2(
             deleteTxns = dnStoreTwoImpl.getDeleteTransactionTable();
         List<DeletedBlocksTransaction> delBlocks = new ArrayList<>();
         int totalBlocks = 0;
+        int blockLimit = 0;
         try (TableIterator<Long,
             ? extends Table.KeyValue<Long, DeletedBlocksTransaction>> iter =
             dnStoreTwoImpl.getDeleteTransactionTable().iterator()) {
-          while (iter.hasNext() && (totalBlocks < blockLimitPerTask)) {
+          while (iter.hasNext() && (blockLimit
+              < blockDeletionLimitPerInterval)) {
```

Comment:
blockDeletionLimitPerInterval should be replaced by numBlocksToDelete for the task.

Reply:
Replaced with blocksToDelete